nityam_kumar/PostgresAssignment/src/controller/teacher.ts

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import uuid from "uniqid";
+import { Context } from "vm";
+import * as teacherService from "../services/teacher";
+import {
+  teacherIDSchema,
+  teacherModifySchema,
+} from "../db/validateSchema/helperSchema";
+import teacherSchema from "../db/validateSchema/teacherSchema";
+import paginationSchema from "../db/validateSchema/paginationSchema";
+import AppError from "../utils/appError";
+
+interface ITeacher {
+  age: number;
+  fname: string;
+  lname: string;
+  teacher_id?: string;
+  sex?: string;
+}
+
+export const createTeacher = async (ctx: Context) => {
+  const t1: ITeacher = ctx.request.body;
+
+  try {
+    await teacherSchema.validateAsync(t1);
+  } catch (err) {
+    throw new AppError(err.message, 400);
+  }
+
+  t1.teacher_id = uuid();
+  await teacherService.addTeacher(t1);
+  ctx.status = 200;
+  ctx.body = {
+    status: `successfully created teacher with ${t1.teacher_id}`,
+  };
+};
+
+export const modifyTeacher = async (ctx: Context) => {
+  const teacher_id = ctx.params.teacherId;
+  const { fname, lname, age } = ctx.request.body;
+
+  try {
+    await teacherModifySchema.validateAsync({ fname, lname, age, teacher_id });
+  } catch (err) {
+    throw new AppError(err.message, 400);
+  }
+
+  await teacherService.checkExists(teacher_id);
```

Comment:
same as above ???

Reply:
await teacherService.checkExists(teacher_id);
  await teacherService.modifyTeacher(fname, lname, age, teacher_id);

before updating teacher info it will check for its existence if not present returns false and proper error msg is displayed

Reply:
code changed